pkg/querier/blocks_scanner.go

```diff
@@ -210,21 +218,16 @@ func (d *BlocksScanner) scanBucket(ctx context.Context) (returnErr error) {
 		}()
 	}
 
-	// Iterate the bucket to discover users.
-	err := d.bucketClient.Iter(ctx, "", func(s string) error {
-		userID := strings.TrimSuffix(s, "/")
+	// Push a job for each user whose blocks need to be discovered.
+	for _, userID := range userIDs {
 		select {
 		case jobsChan <- userID:
-			return nil
+		// Nothing to do.
 		case <-ctx.Done():
-			return ctx.Err()
+			resMx.Lock()
```

Comment:
This will add many context errors to `resErrs` (one per remaining user ID). Previously only one was added, which looks like enough.

Reply:
So glad you spotted it, thanks! Should be fixed now.